lib/msal-core/test/UserAgentApplication.spec.ts

```diff
@@ -2838,6 +2844,33 @@ describe("UserAgentApplication.ts Class", function () {
             expect(acquireTokenSilentPromise instanceof Promise).to.be.true;
             acquireTokenSilentPromise.catch(error => {});
         });
+
+        it("acquireTokenSilent returns even if a new UserAgentApplication instance is instantiated", (done) => {
+            // Tests a bug whereby the instantiation of a 2nd UserAgentApplication was causing in-progress acquireTokenSilent calls on other instances to hang
+            const request = {
+                scopes: [TEST_CONFIG.MSAL_CLIENT_ID], 
+                loginHint: TEST_LOGIN_HINT,
+                forceRefresh: true
+            }
+            const TEST_LIBRARY_STATE_SILENT = RequestUtils.generateLibraryState(Constants.interactionTypeSilent);
+            sinon.stub(RequestUtils, "validateAndGenerateState").returns(TEST_LIBRARY_STATE_SILENT + "|");
+            sinon.stub(CryptoUtils, "createNewGuid").returns("123523")
```

Comment:
Does it matter if this isn't a valid guid, or does this just pad with 0s?

Reply:
It doesn't matter, just needs to be a string. In this case it's forcing acquireTokenSilent to set the nonce to "123523" because that's what our test IdToken has as its nonce.